query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java

```diff
@@ -467,10 +480,39 @@ private void processComputeIfAbsentCommand(final ComputeIfAbsentCommand command,
     */
    private void processComputeIfAbsentCommand(final ComputeIfAbsentCommand command, final InvocationContext ctx, final Object prevValue, final Object computedValue, TransactionContext transactionContext) {
       if (command.isSuccessful()) {
-         processComputes(command, ctx, prevValue, computedValue, transactionContext);
+         processFunctionalCommand(command, ctx, prevValue, computedValue, transactionContext);
       }
    }
-   private void processComputes(AbstractDataWriteCommand command, InvocationContext ctx, Object prevValue, Object computedValue, TransactionContext transactionContext) {
+
+   /**
+    * Indexing management of a ReadWriteKeyCommand
+    *
+    * @param command the ReadWriteKeyCommand
+    * @param ctx the InvocationContext
+    * @param prevValue the previous value on this key
+    * @param transactionContext Optional for lazy initialization, or reuse an existing context.
+    */
+   private void processReadWriteKeyCommand(final ReadWriteKeyCommand command, final InvocationContext ctx, final Object prevValue, TransactionContext transactionContext) {
+      if (command.isSuccessful()) {
+         processFunctionalCommand(command, ctx, prevValue, ctx.lookupEntry(command.getKey()).getValue(), transactionContext);
+      }
+   }
+   /**
+    * Indexing management of a ReadWriteKeyCommand
+    *
+    * @param command the ComputeIfAbsentCommand
+    * @param ctx the InvocationContext
+    * @param prevValue the value before the call
+    * @param resultValue the result of the command call
+    * @param transactionContext Optional for lazy initialization, or reuse an existing context.
+    */
+   private void processReadWriteKeyCommand(final ReadWriteKeyCommand command, final InvocationContext ctx, final Object prevValue, final Object resultValue, TransactionContext transactionContext) {
+      if (command.isSuccessful()) {
+         processFunctionalCommand(command, ctx, prevValue, resultValue, transactionContext);
+      }
+   }
+
+   private void processFunctionalCommand(AbstractDataWriteCommand command, InvocationContext ctx, Object prevValue, Object computedValue, TransactionContext transactionContext) {
```

Comment:
@gustavonalle Upon write, the index for the given entity is invalidated (allegedly :)). But since you don't know the type, you need to do that to all the poor entities.
The indexing is done on all *write* owners. However a node may temporarily be only write owner, not a read owner, and that's the issue with entry not being looked up.